Cesium3DTilesSelection/src/TilesetContentManager.cpp

```diff
@@ -1072,6 +1077,37 @@ bool TilesetContentManager::tileNeedsLoading(const Tile& tile) const noexcept {
          anyRasterOverlaysNeedLoading(tile);
 }
 
+void TilesetContentManager::tickMainThreadLoading(
+    double timeBudget,
+    const TilesetOptions& tilesetOptions) {
+  CESIUM_TRACE("TilesetContentManager::tickMainThreadLoading");
+
+  // A budget of 0.0 indicates that all ready tiles should finish loading.
+  if (timeBudget == 0.0) {
+    for (MainThreadLoadTask& task : this->_finishLoadingQueue) {
+      finishLoading(*task.pTile, tilesetOptions);
+    }
+    return;
```

Comment:
The queue isn't cleared in this case.

Reply:
It looks like tiles only get pushed into `_finishLoadingQueue` when the time budget is not 0.0. But in that case, this loop is unnecessary, right?

Reply:
Yup good point, removed it and added a clarifying comment!